Week 05/README.md

```diff
@@ -169,111 +169,127 @@ const size_t ARRAY_SIZE = 1024;
 int arr[ARRAY_SIZE];
 
 size_t n;
+std::cout << "Enter number of elements: ";
 std::cin >> n;
 
-if(!std::cin)
+if (!std::cin)
 {
-    std::cerr << "Could not read number of elements.";
-    return 1;
+	std::cerr << "Could not read number of elements.";
+	return 1;
 }
 
-if(n > ARRAY_SIZE)
+if (n > ARRAY_SIZE)
 {
-    std::cerr << "Invalid number of elements. Max number of elements is " << ARRAY_SIZE;
-    return 2;
+	std::cerr << "Invalid number of elements. Max number of elements is " << ARRAY_SIZE;
+	return 2;
 }
 
-for(size_t i = 0; i < n; i++)
+std::cout << "Enter the elements: ";
+for (size_t i = 0; i < n; i++)
 {
-    std::cin >> n;
+	std::cin >> arr[i];
 }
 
-if(std::cin.bad())
+if (std::cin.bad())
 {
-    std::cerr << "Could not read elements of array.";
-    return 3;
+	std::cerr << "Could not read elements of array.";
+	return 3;
 }
 
 int search;
+std::cout << "Enter element to search: ";
 std::cin >> search;
 
-if(!std::cin)
+if (!std::cin)
 {
-    std::cerr << "Could not read the searched number.";
-    return 4;
+	std::cerr << "Could not read the searched number.";
+	return 4;
 }
 
 int sum = 0;
-size_t indexFound = ARRAY_SIZE; // invalid index of the array
-for(size_t i = 0; i < n; i++)
+size_t indexFound = -1; // searches for last index with the number
+for (size_t i = 0; i < n; i++)
 {
-    sum += arr[i];
-    if(indexFound != ARRAY_SIZE && arr[i] == searched)
-    {
-        indexFound = i;
-    }
+	sum += arr[i];
+	if (arr[i] == search)
```

Comment:
This was intended to find the index of the first occurrence. Please keep that logic